lib/fastlane/plugin/wpmreleasetoolkit/helper/filesystem_helper.rb

```diff
@@ -36,7 +36,8 @@ def self.plugin_root
         while continue
           child_filenames = dir.children.map! { |x| File.basename(x) }
 
-          if child_filenames.include? 'fastlane-plugin-wpmreleasetoolkit.gemspec'
+          # The first option is for development (where the .gemspec is present in the project root), the second is for production (where it is not)
+          if child_filenames.include?('fastlane-plugin-wpmreleasetoolkit.gemspec') || File.basename(dir).include?('fastlane-plugin-wpmreleasetoolkit')
```

Comment:
Alternatively, you could make `File.basename` on line 37 remove the file extension by using `dir.children.map { |x| File.basename(x, '.gemspec') }`, then this line would simplify to just this:
```suggestion
          # The first option is for development (where the .gemspec is present in the project root), the second is for production (where it is not)
          if child_filenames.include?('fastlane-plugin-wpmreleasetoolkit')
```

Reply:
This made me realize that this code as well as the one for `project_root` both use `map!` (which is the version that mutates-in-place) instead of `map`. It doesn't really makes sense to mutate `dir.children` in place especially because this is the result of a function / property, and not a variable, so I'm actually surprised that Ruby doesn't crash telling you you're trying to mutate a constant value there on lines 37 and 18…

Reply:
Ok I think what confused me during last review (see EDIT in comment above) is that you use `include?` to compare the `File.basename(dir)` and the expected dir name. Like if `File.basename(dir)` was returning an array or list of files, while it's returning a `String`. So your current code would also yield true if the dir was named `fastlane-plugin-wpmreleasetoolkit-backup` as `include?` would match that name. Probably better to use a simple `==` here.
```suggestion
          # The first option is for development (where the .gemspec is present in the project root), the second is for production (where it is not)
          if child_filenames.include?('fastlane-plugin-wpmreleasetoolkit.gemspec') || File.basename(dir) == 'fastlane-plugin-wpmreleasetoolkit'
```

Reply:
Sorry for the slow followup – so your edit was correct! Different behaviour in production vs development.

In development, we'll hit the `.gemspec` path, but in production the path looks like `vendor/bundle/ruby/2.6.0/gems/fastlane-plugin-wpmreleasetoolkit-1.3.0`, hence the `include` (though `starts_with` might be _technically_ safer?)

WDYT?

Reply:
Ahhh yeah makes more sense now. Yeah I think `start_with?` would make more sense then (and maybe even `start_with?('fastlane-plugin-wpmreleasetoolkit-')`? 🤷 )

I think it's worth improving the comment on that line a bit to explain that better